http/src/handler.rs

```diff
@@ -311,13 +343,57 @@ impl<M: Metadata, S: Middleware<M>> RpcHandler<M, S> {
 			Method::Options => {
 				RpcHandlerState::Writing(Response::empty())
 			},
+			// Respond to health API request if there is one configured.
+			Method::Get if self.health_api.as_ref().map(|x| &*x.0) == Some(request.uri().path()) => {
+				RpcHandlerState::ProcessHealth {
+					metadata,
+					method: self.health_api.as_ref()
+							.map(|x| x.1.clone())
+							.expect("Health api is defined since the URI matched."),
+				}
+			},
 			// Disallow other methods.
 			_ => {
 				RpcHandlerState::Writing(Response::method_not_allowed())
 			},
 		}
 	}
 
+	fn process_health(
+		&self,
+		method: String,
+		metadata: M,
+	) -> Result<RpcPollState<M, S::Future>, hyper::Error> {
+		use self::core::types::{Call, MethodCall, Version, Params, Request, Id, Output, Success, Failure};
+
+		// Parse request
+		let call = Request::Single(Call::MethodCall(MethodCall {
+			jsonrpc: Some(Version::V2),
+			method,
+			params: Params::None,
+			id: Id::Num(1),
```

Comment:
can't it clash with actual requests?

Reply:
It doesn't really matter - `id` is only used by the client to correlate request and response. Here:
1. We are in HTTP context so we have correlation already
2. We actually have a `Future` for the response as well, so we don't need `id`
3. `jsonrpc-core` is not using `id ` for anything, it's not even passed to handlers.
4. It's not part of the response at all, so it's just internal quirk to use existing types & methods.

Reply:
but can't client mess responses if he gets two with id="1"?

Reply:
(one that he sent himself, one from health response)